tests/suites/test_suite_ssl.function

```diff
@@ -36,10 +36,18 @@ void log_analyzer( void *ctx, int level,
     }
 }
 
+/* Defaults for min/max version tests (used for arguments as well) */
+#define MBEDTLS_MIN_MAX_VERSION_NONE -1
+#define MBEDTLS_EXPECT_VERSION_NEGOTIATION_FAIL -1
```

Comment:
I was attempting to be deliberately explicit, so that the test definitions in the .data made more sense if you didn't know that NONE meant you were expecting failure. I didn't like the double define either, but I am also not the biggest fan of obfuscation like that.

Reply:
I am not completely sure about what you mean by obfuscation but the last parameter of handshake_version() test function is "expected_version" thus passing a NONE value when no version is expected (negotiation failure) does not seem too wrong to me. For sure when reading test parameters, if you don't know that the last parameter of handshake_version() is the expected version then it is not obvious from the test parameters that the version negotiation is expected to fail. There is already some comments to help there but if you prefer to make it clear at the parameter level then I 'd suggest to add another parameter "expected_result" that would be equal to 0 when the negotiation is expected to succeed and  MBEDTLS_ERR_SSL_BAD_HS_PROTOCOL_VERSION otherwise. This seems a more generic approach to me.

Reply:
Ok, clearly a bad choice of words - what I meant was that the test definition would be readable without needing to look at the function definition. I will implement this combined with the next comment, i.e. one macro for both, renamed.